### PR TITLE
#2084 fix: assign single line height on truncated url line

### PR DESCRIPTION
**Pull request by baproactive**
#2084

### CODE REVIEW

**Review by raineorshine — CHANGES_REQUESTED** [review]
First, start with describing the cause of the bug. *Why* is the height incorrect? It is hard for me to evaluate if the solution is appropriate without knowing the cause of the bug. 

I am guessing that `useMultiline` is returning true for the multiline url, and stays that way even when the url is ellipsized. If so, wouldn't we want to correct the problem there? Hard coding the height would seem to only mask the problem.

**Comment by baproactive** [comment]
> First, start with describing the cause of the bug. _Why_ is the height incorrect? It is hard for me to evaluate if the solution is appropriate without knowing the cause of the bug.
> 
> I am guessing that `useMultiline` is returning true for the multiline url, and stays that way even when the url is ellipsized. If so, wouldn't we want to correct the problem there? Hard coding the height would seem to only mask the problem.

It's not affected by multiline flag, it's affected because of the display:inline-block applied to .single-line .thought-annotation .editable-annotation-text
browser calculates the height differently between display:inline-block and display: inline. 

For those annotation boxes under .single-line class name, they have display:inline-block to apply the css text sllipsis style.

**Review by raineorshine — CHANGES_REQUESTED** [review]
Got it, thanks. Let's add an inline comment explaining that. 

Where is the other `1.9em` defined? It might be a good idea to define that in one place if they are tightly coupled. Otherwise that relationship is not explicit and it's more likely that one might be changed accidentally.

File: src/App.css [hunk 1]
```diff
@@ -630,6 +630,15 @@ This prevents table cols being rendered as multiline content, which causes inaac
   min-height: 1.9em;
 }
 
+/** Assign annotation height on single line truncated url */
+.single-line .thought {
+  /**
+    .singl-line element has display: inline-block property, it affects the height calculation of the .thougt div so we need to set the height manually.
```
Comment (raineorshine) [inline src/App.css:636]: Good comment, thanks.

File: src/App.css [hunk 2]
```diff
@@ -630,6 +630,15 @@ This prevents table cols being rendered as multiline content, which causes inaac
...
+    This height 1.9em is the same value as the min-height above: line: 630
```
Comment (raineorshine) [inline src/App.css:637]: We should avoid specifying exact line numbers in comments, as they are quickly invalidated if any code is added or removed above.

File: src/App.css [hunk 3]
```diff
@@ -630,6 +630,15 @@ This prevents table cols being rendered as multiline content, which causes inaac
...
+  */
+  height: 1.9em;
```
Comment (raineorshine) [inline src/App.css:639]: We can leverage CSS variables for a more explicit enforcement of using the same value across multiple properties.

**Review by raineorshine — APPROVED** [review]

**Pull request merged**